### ISSUE TITLE
kubectl explain openapiv3: Future Improvements

### ISSUE THREAD

**Issue by alexzielenski** [kubernetes/kubernetes#113958]
### What would you like to be added?

- [x] https://github.com/kubernetes/kubernetes/issues/113903#issuecomment-1317369712
- [ ] Display new v3 `nullable` field. Check if populated by k8s. If not, then create infrastructure to do so.
- [ ] Display new v3 `deprecated` field. Check if populated by k8s. If not, then create infrastructure to do so.
- [ ] Display `examples` field. Maybe explore adding examples output?
- [ ] Display `default` field. Maybe for strings below certain length/ints/bools could be shown.
- [x] Turn into Beta, deprecate old `kubectl explain`, have a fallback mechanism

### Why is this needed?

- KEP: https://github.com/kubernetes/enhancements/issues/3515

**Comment by alexzielenski** [comment]
whoops

/sig cli
/remove-sig api-machinery

**Comment by yashsingh74** [comment]
@alexzielenski If the issue is still open, may I help in any way for the completion of this task?

**Comment by alexzielenski** [comment]
@yashsingh74 definitely! How would you like to take the task of populating the `nullable` information in kubectl explain output.

The output is generated by Go template from the OpenAPIV3 json response.

Go template is here: https://github.com/kubernetes/kubernetes/blob/d7bff1c80912b20bc969111ac52825b67909bbe8/staging/src/k8s.io/kubectl/pkg/explain/v2/templates/plaintext.tmpl

Please feel free to open a WIP PR which refers to this one in the description once you've gotten started.

**Comment by atiratree** [comment]
@alexzielenski I have found an [issue](https://github.com/kubernetes/kubernetes/issues/118875) with kubectl explain. I am trying to fix it here: https://github.com/kubernetes/kubernetes/pull/118876
